### PR TITLE
feat: check for files above limit first and then check for conflicts

### PR DESCRIPTION
**Pull request by thiagocarvalhodev**
Changes the order of validation from:

**name conflicting → check file size** 

to

**check file size → name conflicting**

### CODE REVIEW

**Review by karlprieb — DISMISSED** [review]

File: lib/components/upload_form.dart [hunk 1]
```diff
@@ -219,10 +221,10 @@ class UploadForm extends StatelessWidget {
                 ),
                 if (state.hasFilesToUpload)
                   TextButton(
-                    onPressed: () => context
-                        .read<UploadCubit>()
-                        .prepareUploadPlanAndCostEstimates(
-                            uploadAction: UploadActions.SkipBigFiles),
+                    onPressed: () {
+                      context.read<UploadCubit>().removeBigFiles();
+                      context.read<UploadCubit>().checkConflicts();
```
Comment (fedellen) [inline lib/components/upload_form.dart:226]: This second call is async, is it okay to not await here?

File: test/blocs/upload_cubit_test.dart [hunk 2]
```diff
@@ -374,9 +374,10 @@ void main() {
             tNoConflictingFiles..add(tTooLargeFile));
       },
       act: (cubit) async {
+        // TODO(@thiagocarvalhodev): Review
         await cubit.startUploadPreparation();
         await cubit.prepareUploadPlanAndCostEstimates(
-            uploadAction: UploadActions.SkipBigFiles);
+            uploadAction: UploadActions.Skip);
```
Comment (fedellen) [inline test/blocs/upload_cubit_test.dart:380]: We didn't really get any test coverage over the new logic here. UploadReady will have the same behavior as before right?

File: lib/blocs/upload/upload_cubit.dart [hunk 3]
```diff
@@ -191,7 +211,7 @@ class UploadCubit extends Cubit<UploadState> {
           : '${_targetFolder.path}/${folder.name}';
     }
     final filesToUpload = <WebFile>[];
-    files.forEach((file) {
+    for (var file in files) {
```
Comment (fedellen) [inline lib/blocs/upload/upload_cubit.dart:214]: Is this a performance increase for Dart?

**Review by fedellen — APPROVED** [review]
Left some non-blocking questions. Looking great!

File: lib/components/upload_form.dart (hunk 1 above)
Comment (thiagocarvalhodev) [reply lib/components/upload_form.dart:226]: It's okay! As this function changes the states of this page, we don't need to wait as at every state change this page will be affected by the side effects.

File: test/blocs/upload_cubit_test.dart (hunk 2 above)
Comment (thiagocarvalhodev) [reply test/blocs/upload_cubit_test.dart:380]: Yeah. Those tests should be refactored. To test those functions today is currently not possible due to the upload_cubit depends of other classes that import `dart:html` directly, so the tests not compile. If you are able to compile, please let me know!

File: lib/blocs/upload/upload_cubit.dart (hunk 3 above)
Comment (thiagocarvalhodev) [reply lib/blocs/upload/upload_cubit.dart:214]: It is not, it is just to fix the linter rule: `Avoid using forEach with a function literal.` 

But I agree, it should be a final variable as we aren't changing it!

**Pull request merged**